fili-core/src/main/java/com/yahoo/bard/webservice/logging/LogFormatterProvider.java

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.logging;
+
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A class to provide a log formatter instance.
+ */
+public class LogFormatterProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SystemConfigProvider.class);
+
+    private static final String LOG_FORMATTER_IMPL_KEY = "fili__log_formatter_impl";
+
+    private static final String DEFAULT_LOG_FORMATTER_IMPL = JsonLogFormatter.class.getCanonicalName();
+
+    /**
+     * The instance of the Log Formatter.
+     */
+    private static final LogFormatter LOG_FORMATTER = getInstance();
+
+    /**
+     *  Get an instance of LogFormatter.
+     *
+     *  @return an instance of LogFormatter
+     */
+    public static LogFormatter getInstance() {
+        if (LOG_FORMATTER != null) {
+            return LOG_FORMATTER;
+        }
+        String logFormatterImplementation = System.getenv(LOG_FORMATTER_IMPL_KEY);
+        try {
+            if (logFormatterImplementation == null) {
+                logFormatterImplementation = System.getProperty(LOG_FORMATTER_IMPL_KEY, DEFAULT_LOG_FORMATTER_IMPL);
```

Comment:
I think here you should actually be using an instance of `SystemConfig`:

```java
logFormatterImplementation = SystemConfigProvider.getInstance().getStringProperty(
        SYSTEM_CONFIG.getPackageVariableName(LOG_FORMATTER_IMPL_KEY),
        DEFAULT_LOG_FORMATTER_IMPL
);
```

Also, I _think_ line `35` can be removed, and we can drop the `null` check.

@michael-mclawhorn Could probably can verify that for me.

The reason the `SystemConfigProvider` accesses the environment directly, is because it can't very well ask itself about its configuration parameter!

Everything else should go through the `SystemConfig` object though.